Only use GDM_LANG if not empty

gdm >= 2.91 sets GDM_LANG to an empty string which xfce-session
ends up using, overriding existing good LANG and breaking up utf8 etc.

diff -up xfce4-session-4.8.1/xfce4-session/main.c.gdmlang xfce4-session-4.8.1/xfce4-session/main.c
--- xfce4-session-4.8.1/xfce4-session/main.c.gdmlang	2011-03-10 15:59:08.000000000 +0200
+++ xfce4-session-4.8.1/xfce4-session/main.c	2011-03-10 15:59:30.000000000 +0200
@@ -103,7 +103,7 @@ setup_environment (void)
 
   /* this is for compatibility with the GNOME Display Manager */
   lang = g_getenv ("GDM_LANG");
-  if (lang != NULL)
+  if (lang != NULL && strlen (lang) > 0)
     {
       xfce_setenv ("LANG", lang, TRUE);
       xfce_unsetenv ("GDM_LANG");
